### DISCUSSION TITLE
Remember last window position and size

### DISCUSSION THREAD

**Discussion by evertonstz** [ghostty-org/ghostty#4204] [Feature Requests, Ideas] [answered]
I'm one of those guys that keeps closing and reopening their terminal during the day and that's one of the first things I noticed when using Ghostty: it doesn't remembers the size and position of the last closed window like most other MacOS apps.
It's been a long time since I used Linux with modern gnome, but I think that's also the expected behavior for apps there, right?

**Comment by rhodes-b** [comment]
This is not the default behavior of apps on linux some do it but its not the standard

iirc no GTK default apps do this (tested with gnome-term, nemo, and xed), if there is an app on linux that does its custom behavior as GTK does not provide this

**Reply by mitchellh** [reply]
We should do this only on macOS.

**Reply by Nepherim** [reply]
Why should the size not be remembered on Linux? Plenty of Gnome apps remember the previous windows size. The app store for one.

**Reply by 00-kat** [reply]
Open a new Discussion (assuming one doesn't exist already) instead of bumping this one.

**Reply by ihpled** [reply]
> Why should the size not be remembered on Linux? Plenty of Gnome apps remember the previous windows size. The app store for one.

For bash shell:
- add those lines in ghostty config file (use two arbitrary initial values):
```
window-width = 80
window-height = 40
```
- add this code at the end of your .bashrc
```
save_ghostty_dimensions() {
    if [ "$TERM_PROGRAM" = "ghostty" ]; then
        read -r rows cols <<< "$(stty size)"
        config_file="$HOME/.config/ghostty/config"

        if [ -f "$config_file" ]; then
            sed -i "s/^window-width = .*/window-width = $cols/" "$config_file"
            sed -i "s/^window-height = .*/window-height = $rows/" "$config_file"
        fi
    fi
}

trap save_ghostty_dimensions EXIT
```
For fish shell:
- add this to ~/.config/fish/conf.d/ghostty_save.fish
```
function save_ghostty_dimensions --on-event fish_exit
    if test "$TERM_PROGRAM" = "ghostty"
        stty size | read -l rows cols
        set -l config_file ~/.config/ghostty/config

        if test -f $config_file
            sed -i "s/^window-width = .*/window-width = $cols/" $config_file
            sed -i "s/^window-height = .*/window-height = $rows/" $config_file
        end
    end
end
```
Now closing the console window your shell will save current dimensions in ghostty config file.

**Comment by mitchellh** [comment, marked as answer]
Approved #4233

**Comment by StudioCrash** [comment]
These solutions, along with the built-in ghostty config file settings  ```
window-width = 110
window-height = 40``` for example, remember size but not window placement.  At least I know this is true in Linux with Gnome.  I know this is a nice-to-have issue, but, it, would be nice.  :-)

**Reply by zquestz** [reply]
Agreed, adding a way to save window position would be extremely nice. Other terminals for Linux have this, for example Kitty.
